tools/topo-lookup.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+
+# Import the Compute Engine API
+# pip install google-cloud-compute 
+from google.cloud import compute_v1
+from google.cloud.compute_v1 import Instance
+
+DESCRIPTION = """
+topo-lookup will lookup physical hosts and report their distribution.
+Usage:
+tools/topo-lookup.py --project_id=ns-playground-2023-01-19 --zone=us-central1-a --name_re=.*my-vm.*
+"""
+
+# Create a Compute Engine API client
+client = compute_v1.InstancesClient()
+
+def lookup_vm_matches(name_re: str, zone: str, project_id: str) -> list[Instance]:
+    """Looks up all vm instances with a given name pattern."""
+    request = compute_v1.ListInstancesRequest(
+        project=project_id, zone=zone, max_results=10000, filter=f"name eq {name_re}"
+    )
+    return client.list(request)
+
+def analyze(instances: list[Instance]) -> tuple:
```

Comment:
What type check did you use?
`tuple -> Tuple[int, int, int]`

Reply:
Python > Analysis: Type Checking Mode = basic
I believe it is backed by pylance extension, or maybe just python extension

I had it on basic setting. If I turn it to strict then it warns on this (I have updated) but it also annoyingly warns on like every single variable in the whole file, so I leave it on basic. 

Do you have a suggestion?

Reply:
Land it as is. It invites a broader discussion that we can have as a team offline.